### PR TITLE
#566 Enable Grafeas to store and retrieve Vex information

### PR DESCRIPTION
**Pull request by yeshwanth1993**
1. Adds a VulnerabilityAssessment Note.
2. Adds a VexAssessment field to the vulnerability occurrence proto.
3. Generates relevant go protobuf files

### CODE REVIEW

File: proto/v1beta1/swagger/merged/grafeas.swagger.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
           "default": {
             "description": "An unexpected error response.",
             "schema": {
-              "$ref": "#/definitions/rpcStatus"
+              "$ref": "#/definitions/googlerpcStatus"
```
Comment (wkozlik) [inline proto/v1beta1/swagger/merged/grafeas.swagger.json:36]: Is this potentially a breaking change? Where is it coming from?

File: proto/v1beta1/swagger/merged/grafeas.swagger.json [hunk 2]
```diff
@@ -1450,6 +1556,29 @@
       },
       "description": "A note that indicates a type of analysis a provider would perform. This note\nexists in a provider's project. A `Discovery` occurrence is created in a\nconsumer's project at the start of analysis."
     },
+    "googlerpcStatus": {
```
Comment (wkozlik) [inline proto/v1beta1/swagger/merged/grafeas.swagger.json:1559]: I am curious what caused the change of the rpcStatus to googlerpcStatus.

**Review by wkozlik — CHANGES_REQUESTED** [review]

File: proto/v1beta1/swagger/merged/grafeas.swagger.json (hunk 2 above)
Comment (yeshwanth1993) [reply proto/v1beta1/swagger/merged/grafeas.swagger.json]: I don't know how this happened tbh. I regenerated everything and it looks good now.

File: proto/v1beta1/vex.proto [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    // Token that identifies a product so that it can be referred to from other
+    // parts in the document. There is no predefined format as long as it
+    // uniquely identifies a group in the context of the current document.
+    string id = 2;
+
+    oneof identifier {
+      // Contains a URI which is vendor-specific.
+      // Example: The artifact repository URL of an image.
+      grafeas.v1beta1.RelatedUrl  generic_uri = 3;
```
Comment (wkozlik) [inline proto/v1beta1/vex.proto:83]: Why to use RelatedUrl instead of string? It seems like a stretch.

File: proto/v1beta1/vulnerability.proto [hunk 4]
```diff
@@ -202,7 +203,39 @@ message Details {
...
+    // tracking number for the vulnerability.
+    string cve = 1;
+
+    // Holds a list of references associated with this vulnerability item and
+    // assessment.
+    repeated grafeas.v1beta1.RelatedUrl related_uris = 2;
+
+    // The VulnerabilityAssessment note from which this VexAssessment was
+    // generated.
+    string note = 3;
```
Comment (wkozlik) [inline proto/v1beta1/vulnerability.proto:219]: To be consistent, should this be renamed to `note_name` ?

**Review by wkozlik — CHANGES_REQUESTED** [review]

**Review by wkozlik — APPROVED** [review]
LGTM

**Pull request merged**